core/core_integrations/messenger.js

```diff
@@ -0,0 +1,198 @@
+	}
+	
+	return spl;
+},
+
+genericMessageRequest = function(message, recipientId, callback) {
+	request({
+		url: 'https://graph.facebook.com/v2.6/me/messages',
+		qs: {access_token: exports.config.token},
+		method: 'POST',
+		json: {
+			recipient: {id: recipientId},
+			message: message,
+		}
+	},
+	function(error, response, body) {
+		if (error) {
+			throw error;
+		}
+		if (callback) {
+			callback();
+		}
+	});
+},
+
+listGenericMessageRequest = function(list, property, threadId) {
+	if (!list || list.length <= 0) {
+		return;
+	}
+	var arg = {};
+	arg[property] = list[0];
+	list.splice(0, 1);
+	genericMessageRequest(arg, threadId, listGenericMessageRequest.bind(this, list, property, threadId));
+};
+	
+exports.start = function(callback) {
+	api = shim.createPlatformModule({
+		sendMessage: function (message, threadId) {
+			var spl = sentenceSplitter(message);
+			listGenericMessageRequest(spl, 'text', threadId);
+		},
+		getUsers: function() {
+			return exports.config.users || {};
+		},
+		sendImage: function(type, image, description, thread) {
+			if (description) {
+				this.sendMessage(description, thread);
+			}
+			switch(type) {
+				case 'url':
+					genericMessageRequest({
+						attachment: {
+							type: 'image',
+							payload: {
+								url: image
+							}
+						}
+					}, thread);
+					break;
+				case 'file':
+					genericMessageRequest({
+						attachment: {
+							type: 'image',
+							payload: image
+						}
+					}, thread);
+					break;
+			}
+		},
+		sendPrivateMessage: this.sendMessage,
+		commandPrefix: exports.config.commandPrefix
+	});
+	
+	var app = express();
+	app.use(bodyParser.urlencoded({extended: false}));
+	app.use(bodyParser.json());
+
+	var port = exports.config.port;
+	if (exports.config.cert && exports.config.ca && exports.config.key) {
+		var sslOpts = {
+			key: fs.readFileSync(path.resolve(exports.config.key), 'utf8'),
+			cert: fs.readFileSync(path.resolve(exports.config.cert), 'utf8'),
+			ca: fs.readFileSync(path.resolve(exports.config.ca), 'utf8')
+		};
+		http = https.createServer(sslOpts, app);
+		port = port || 443;
+	}
+	else {
+		http = require('http').createServer(app);
+		port = port || 80;
+	}
+	
+	http.listen(port);
+	
+	app.get('/', (req, res) => {
+		res.send('Invalid Request');
+	});
+	
+	// Verify Webhook
+	app.get('/webhook', function (req, res) {
+		if (req.query['hub.verify_token'] === exports.config.verify) {
+			res.send(req.query['hub.challenge']);
+		} else {
+			res.send('Invalid verify token');
+		}
+	});
+	
+	// Received Post
+	app.post('/webhook', function (req, res) {
+		res.sendStatus(200);
+		var msg = req.body.entry[0],
+			userId = msg.messaging[0].sender.id;
+		
+		if (!msg.messaging[0].message) {
+			return;
+		}
+		
+		getNameForId(userId, function(name) {
+			var event = shim.createEvent(userId, userId, name, msg.messaging[0].message.text + '');
+			callback(api, event);
+		});
+	});
+};
+
+exports.stop = function() {
+	http.close();
+};
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2554612143/issues/source?bid=3282987&fileBranchId=3288129#l198)